tests/lib/rules/id-length.js

```diff
@@ -0,0 +1,66 @@
+    ESLintTester = require("eslint-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var eslintTester = new ESLintTester(eslint);
+eslintTester.addRuleTest("lib/rules/id-length", {
+    valid: [
+        "var xyz;",
+        "var xy = 1;",
+        "function xyz() {};",
+        "function xyz(abc, de) {};",
+        "var obj = { abc: 1, de: 2 };",
+        "var obj = { 'a': 1, bc: 2 };",
+        "var obj = {}; obj['a'] = 2;",
+        "abc = d;",
+        "try { blah(); } catch (err) { /* pass */ }",
+        "var _ad$$ = new $;",
+        "var xyz = new ΣΣ();",
+        "for (var i=0; i < 10; i++) { console.log(i); }",
+        "for (var _i=0; _i < 10; _i++) { console.log(_i); }",
+        "j=0; while (j > -10) { console.log(--j); }",
+        { code: "var x = Foo(42)", args: [1, {"min": 1}] },
+        { code: "var x = Foo(42)", args: [1, {"min": 0}] },
+        { code: "foo.$x = Foo(42)", args: [1, {"min": 1}] },
+        { code: "var _$xt_$ = Foo(42)", args: [1, {"min": 2, "max": 4}] },
+        { code: "try { blah(); } catch (e) { /* pass */ }", args: [1, {disableFor: ["CatchClause"]}] },
+        { code: "var xy = function x() {};", args: [1, {"min": 2, disableFor: ["FunctionExpression"]}] },
+        { code: "function x() {};", args: [1, {"min": 2, "max": 0, disableFor: ["FunctionDeclaration"]}] },
+        { code: "function xy(a, xyzt) {};", args: [1, {"max": 3, disableFor: ["FunctionDeclaration"]}] },
+        { code: "var ab = function xy(a, xyzt) {};", args: [1, {"max": 3, disableFor: ["FunctionExpression"]}] },
+        { code: "var a; var obj = {a: 1, b: 2, c: a};", args: [1, {disableFor: ["VariableDeclarator", "ObjectExpression"]}] },
+        { code: "for (var q, h=0; h < 10; h++) { console.log(h); q++;}", args: [1, {exceptions: ["h", "q"]}] }
+    ],
+    invalid: [
+        { code: "var x = 1;", errors: [{ message: "Identifier name 'x' is too short. (< 2)", type: "Identifier"}] },
+        { code: "var x;", errors: [{ message: "Identifier name 'x' is too short. (< 2)", type: "Identifier"}] },
+        { code: "function x() {};", errors: [{ message: "Identifier name 'x' is too short. (< 2)", type: "Identifier"}] },
+        { code: "function xyz(a) {};", errors: [{ message: "Identifier name 'a' is too short. (< 2)", type: "Identifier"}] },
+        { code: "var obj = {a: 1, bc: 2};", errors: [{ message: "Identifier name 'a' is too short. (< 2)", type: "Identifier"}] },
+        { code: "try { blah(); } catch (e) { /* pass */ }", errors: [{ message: "Identifier name 'e' is too short. (< 2)", type: "Identifier"}] },
+        { code: "var handler = function (e) {};", errors: [{ message: "Identifier name 'e' is too short. (< 2)", type: "Identifier"}] },
+        { code: "var handler = function ($e) {};", errors: [{ message: "Identifier name '$e' is too short. (< 2)", type: "Identifier"}] },
```

Comment:
These type of errors might be a bit confusing since it says `$e` is too short (< 2), where it actually is two characters and we check the "bare" length after stripping prefixes and suffixes. I couldn't find a succinct enough error message to express this subtlety :(